node/forest_libp2p/src/rpc/blocksync_message.rs

```diff
@@ -0,0 +1,125 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use forest_blocks::BlockHeader;
+use forest_cid::Cid;
+use forest_encoding::{
+    de::{self, Deserialize, Deserializer},
+    ser::{self, Serialize, Serializer},
+};
+use forest_message::{SignedMessage, UnsignedMessage};
+
+/// The payload that gets sent to another node to request for blocks and messages. It get DagCBOR serialized before sending over the wire.
+#[derive(Clone, Debug, PartialEq)]
+pub struct BlockSyncRequest {
+    /// The tipset to start sync from
+    pub start: Vec<Cid>,
+    /// The amount of epochs to sync by
+    pub request_len: u64,
+    /// 1 = Block only, 2 = Messages only, 3 = Blocks and Messages
+    pub options: u64,
+}
+
+impl Serialize for BlockSyncRequest {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        (&self.start, &self.request_len, &self.options).serialize(serializer)
+    }
+}
+impl<'de> Deserialize<'de> for BlockSyncRequest {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let (start, request_len, options) = Deserialize::deserialize(deserializer)?;
+        Ok(BlockSyncRequest {
+            start,
+            request_len,
+            options,
+        })
+    }
+}
+
+/// The response to a BlockSync request.
+#[derive(Clone, Debug, PartialEq)]
+pub struct BlockSyncResponse {
+    /// The tipsets requested
+    pub chain: Vec<TipSetBundle>,
+    /// Error code
+    pub status: u64,
+    pub message: String,
```

Comment:
Just a note about this, the message isn't in the spec so it's possible it will be removed in the future, but it exists in Lotus so we are including for now